infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/KubernetesServerExposer.java

```diff
@@ -290,6 +278,31 @@ private void exposeNonSecureServers(
     }
   }
 
+  private Service createService(
+      Map<String, ServerConfig> internalServers,
+      Map<String, ServerConfig> externalServers,
+      Map<String, ServicePort> unsecuredPorts,
+      Map<String, ServicePort> securedPorts) {
+    Map<String, ServerConfig> allInternalServers = new HashMap<>(internalServers);
+    // Adding external secure services, as they, in fact, internal also.
+    for (ServicePort servicePort : securedPorts.values()) {
+      Map<String, ServerConfig> securedExternalServers = match(externalServers, servicePort);
+      allInternalServers.putAll(securedExternalServers);
```

Comment:
So externally exposed secured servers are added to the list of internal servers. I'm not sure I understand this...

Reply:
Previously we were add all externas servers to the list, and create servers for all of them (which is caused the error we're fixing here). Now it's filtered to secured only (they still need server, to be an upstream for JWT proxy). Or i'm my undestanding is wrong?

Reply:
If we don't need any extrernal servers here at all, i agree :) But i doubt why this line was added by you:
`allNonSecureServers.putAll(externalServers);`?

Reply:
Indeed :)